cuplaConfig.cmake

```diff
@@ -93,6 +93,11 @@ list(APPEND CMAKE_MODULE_PATH "${_cupla_ROOT_DIR}")
 #list(APPEND CMAKE_MODULE_PATH "$ENV{ALPAKA_ROOT}")
 
 
+# C++11 standard for whole project (this is required, even though Alpaka sets C++11 too)
+set(CMAKE_CXX_STANDARD 11)
```

Comment:
these variables must not be set in library code but are set by the application using cupla.

Try setting these as target properties from determined `CMAKE_CXX_STANDARD` (with fallbacks to C++11 if unset).

Reply:
Agree this variable should not be set like that. Maybe an even easier solution without using `CMAKE_CXX_STANDARD` at all: [example 1](https://github.com/robotology/how-to-export-cpp-library/issues/44), [example 2](https://cliutils.gitlab.io/modern-cmake/chapters/features/cpp11.html). This would effectively set a minimum required version: if a host application is setting higher version in `CMAKE_CXX_STANDARD`, that will be used. If a host application does not set it or sets lower version, C++11 will be used for cupla.

Reply:
yeah, you are absolutely right, will look into that.

Reply:
You can also take some details from openPMD-api in which I try to follow recent CMake best-practices: https://github.com/openPMD/openPMD-api/blob/dev/CMakeLists.txt